src/libsyntax/parse/parser.rs

```diff
@@ -640,8 +640,14 @@ impl<'a> Parser<'a> {
         match self.token {
             token::Ident(i) => {
                 if self.token.is_reserved_ident() {
-                    self.span_err(self.span, &format!("expected identifier, found {}",
-                                                      self.this_token_descr()));
+                    let desc = self.this_token_descr();
+                    if desc == "keyword `ref`" {
+                        self.span_err_help(self.span,
```

Comment:
I wondered for a while between the both of them. But if you prefer suggestion, I'll totally update to it.